lib/crawler/link/checker.ex

```diff
@@ -11,14 +10,21 @@ defmodule Crawler.Link.Checker do
   def verify_link(:done, _base_url, _depth) do
     :ok
   end
+
   def verify_link(link, base_url, depth) do
-    url = base_url <> link
-    client = Application.get_env(:crawler, :http_client, PoisonClient)
-    case apply(client, :get, [url, [recv_timeout: 15_000]]) do
-      {:ok, %HTTPoison.Response{status_code: code} = response}
-        when code in 200..399 -> handle_success(link, response, depth, base_url)
-      {:ok, %HTTPoison.Response{status_code: code}} -> handle_error(link, code)
-      {:error, %HTTPoison.Error{reason: reason}}  -> handle_error(link, reason)
+    with url <- URI.merge(base_url, URI.encode(link)),
```

Comment:
is there a scenario where we want to allow either of these to fail or return nil? just wondering if there's really a benefit to using `with` instead of just declaring variables normally.

Reply:
Yeah, I put that in when I was messing around with stuff, but there's no real reason for it so I backed it out.